source/common/common/utility.cc

```diff
@@ -20,8 +21,116 @@
 #include "spdlog/spdlog.h"
 
 namespace Envoy {
+
+namespace {
+
+class SubsecondConstantValues {
+public:
+  const std::string PLACEHOLDER{"?????????"};
```

Comment:
You probably don't need this to be a constant. You can just do something like `std::string(N, '?')` later on to generate a string of length `N`.